### ISSUE TITLE
Middleware response() is not invoked if there was an error in async handler

### ISSUE THREAD

**Issue by mitskevich** [actix/actix-web#187]
First of all, thank you very much for this great library. I've just started to learn how to use it, but I already love it! During my experiments I've found this strange quirk, I'm not sure if it is an actual issue, or something that I'm misunderstanding.

It seems, that `actix-web` doesn't call middleware's `response()` method if there was an error in the async handler.

Example:

```rust
extern crate actix_web;
extern crate futures;

use actix_web as aw;
use futures::Future;
use futures::future;
use actix_web::AsyncResponder;

fn index_result(_req: aw::HttpRequest) -> aw::Result<String> {
    Err(aw::error::ErrorBadRequest("test"))
}

fn index_fut(_req: aw::HttpRequest) -> Box<Future<Item=String, Error=aw::Error>> {
    future::result(Err(aw::error::ErrorBadRequest("test"))).responder()
}

struct SomeMiddleware;

impl<S> aw::middleware::Middleware<S> for SomeMiddleware {
    fn start(&self, _req: &mut aw::HttpRequest<S>) -> aw::Result<aw::middleware::Started> {
        println!("middleware started");
        Ok(aw::middleware::Started::Done)
    }

    fn response(
        &self,
        _req: &mut aw::HttpRequest<S>,
        resp: aw::HttpResponse
    ) -> aw::Result<aw::middleware::Response> {
        println!("middleware response");
        Ok(aw::middleware::Response::Done(resp))
    }
}

fn main() {
    aw::server::new(
        || aw::App::new()
            .middleware(SomeMiddleware)
            .resource("/res/", |r| r.f(index_result))
            .resource("/fut/", |r| r.f(index_fut)))
        .bind("127.0.0.1:8088").expect("Can not bind to 127.0.0.1:8088")
        .run();
}
```

After starting this server, I run the following `curl` command:

```bash
curl -i localhost:8088/fut/
```

I expect to see this in server's stdout:

```
middleware started
middleware response
```

But instead, only first line is displayed.

```
middleware started
```

Curiously, when I invoke handler that uses `Result` as return type, I get the expected output (both lines).

```bash
curl -i localhost:8088/res/
```

I have tested this on `0.5.2` and `master` branch.

**Comment by fafhrd91** [comment]
might be bug, i'll  check

**Comment by fafhrd91** [comment]
fixed by f907be585e0aaad21e1289bdb5c84c27b1a85550

i will release new version tomorrow

**Comment by mitskevich** [comment]
Great, thanks!
